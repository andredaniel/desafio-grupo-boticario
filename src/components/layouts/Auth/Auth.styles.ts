import styled from 'styled-components'
import { theme } from '../../../theme'

export const AuthWrapper = styled.div`
  align-items: center;
  background-color: ${theme.colors.primary.DARK};
  background-image: linear-gradient(
    132deg,
    ${theme.colors.primary.MAIN} 0%,
    ${theme.colors.primary.DARK} 100%
  );
  display: flex;
  justify-content: center;
  min-height: 100vh;
  padding: 50px 20px;
  position: relative;
  
  @media (min-width: 768px) {
    padding: 50px;
  }

  &:after {
    background-image: url('/images/login-background-pattern.png');
    content: '';
    height: 100%;
    left: 0;
    pointer-events: none;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: 1;
  }

  hr {
    background-color: #ccc;
    border: none;
    height: 1px;
    margin: 1.5rem 0;
    width: 100%;
  }
`

export const AuthContent = styled.div`
  align-items: center;
  background-repeat: repeat;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 2rem;
  position: relative;
  z-index: 2;
  order: 2;

  .head {
    margin: 2rem 0;
    text-align: center;
  }

  svg {
    height: auto;
    max-width: 200px;
  }

  h1 {
    color: ${theme.colors.primary.LIGHT};
    font-size: 25px;
    font-weight: normal;
    margin: 0;
  }

  p {
    color: #666;
    margin: 5px 0 0;
  }
`

export const AuthCover = styled.div`
  background-image: url('/images/login-picture.png');
  background-position: 60% center;
  background-size: cover;
  height: 100%;
  width: 100%;
`

export const Card = styled.div`
  background-color: #fff;
  border-radius: 10px;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 200px 3fr;
  margin: 0 auto;
  max-width: 1100px;
  overflow: hidden;
  width: 100%;
  order: 1;
  
  @media (min-width: 768px) {
    grid-template-columns: 2fr 3fr;
    grid-template-rows: 1fr;
  }
`
