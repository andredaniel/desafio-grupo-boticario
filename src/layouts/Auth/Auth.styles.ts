import styled from 'styled-components'
import { Card } from '../../components'
import { theme } from '../../theme'

export const AuthWrapper = styled.div`
  align-items: center;
  background-color: #000;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  min-height: 100vh;
  min-height: calc(var(--vh, 1vh) * 100);
  padding: 50px 20px;
  position: relative;

  @media (min-width: ${theme.breakpoints.md}) {
    flex-direction: row;
    min-height: 100vh;
    padding: 50px;
  }

  svg {
    height: auto;
    margin: auto 0 50px 0;
    max-width: 200px;
    order: 1;
    position: relative;
    z-index: 3;

    @media (min-width: ${theme.breakpoints.md}) {
      margin: 0;
      max-width: 300px;
    }
  }

  &:after {
    background-image: url('/images/login-background-pattern.png');
    content: '';
    height: 100%;
    left: 0;
    pointer-events: none;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: 1;
  }

  hr {
    background-color: ${theme.colors.secondary.CONTRAST};
    border: none;
    height: 1px;
    margin: 1rem 0;
    width: 100%;
  }
`

export const AuthContent = styled.div`
  align-items: center;
  background-repeat: repeat;
  display: flex;
  flex-direction: column;
  justify-content: center;
  order: 2;
  padding: 2rem;

  @media (min-width: ${theme.breakpoints.md}) {
    order: 1;
  }

  .head {
    margin: 0.5rem 0 1.5rem;
    text-align: center;
  }

  h1 {
    color: ${theme.colors.primary.LIGHT};
    font-size: 25px;
    font-weight: normal;
    margin: 0;
  }

  p {
    color: #666;
    margin: 5px 0 0;
  }

  form {
    width: 100%;
  }

  a,
  button {
    width: 100%;
  }
`

export const AuthCover = styled.div`
  background-image: url('/images/pexels-brazilian-smile.jpeg');
  background-position: center;
  background-size: cover;
  height: 100%;
  left: 0;
  opacity: 0.6;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 1;
`

export const StyledCard = styled(Card)`
  display: grid;
  max-width: 500px;
  order: 2;
  overflow: hidden;
  position: relative;
  width: 100%;
  z-index: 2;

  @media (min-width: ${theme.breakpoints.md}) {
    order: 1;
  }
`
