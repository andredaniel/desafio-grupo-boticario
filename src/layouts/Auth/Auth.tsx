import React from 'react'
import { ReactComponent as Brand } from '../../assets/brand/logo-black.svg'
import { AuthContent, AuthCover, AuthWrapper, Card } from './Auth.styles'

const Auth: React.FC = ({ children }): JSX.Element => {
  return (
    <AuthWrapper>
      <Card>
        <AuthContent>
          <Brand />
          {children}
        </AuthContent>
        <AuthCover />
      </Card>
    </AuthWrapper>
  )
}

export { Auth }
