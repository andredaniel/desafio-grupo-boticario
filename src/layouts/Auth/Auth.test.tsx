import { render } from '@testing-library/react'
import React from 'react'
import { Auth } from './Auth'

describe('<Auth />', () => {
  it('should render Auth component properly', () => {
    const { container } = render(<Auth />)
    expect(container).toMatchSnapshot()
  })
})
