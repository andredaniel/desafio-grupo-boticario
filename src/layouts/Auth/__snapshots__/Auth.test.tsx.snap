// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> should render Auth component properly 1`] = `
<div>
  <div
    class="sc-gKsewC iELeXN"
  >
    <div
      class="sc-dlfnbm yYWWd sc-pFZIQ iuJdM"
    >
      <div
        class="sc-iBPRYJ cIUzNH"
      />
    </div>
    <svg>
      logo-white.svg
    </svg>
    <div
      class="sc-fubCfw fmEiLQ"
    />
  </div>
</div>
`;
